junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java

```diff
@@ -153,6 +159,37 @@ private RunListener createRunListener(String runListenerClassName) {
     } else return null;
   }
 
+  /**
+   * Alter default RunSettings depending on the passed selectors. If selectors contains only
+   * elements of type TestSelector, then default settings are altered to include only test names
+   * from these selectors. This allows to run particular test cases within given test class.
+   * testFilter is treated as a regular expression, hence joining is done via '|'.
+   */
+  private RunSettings alterRunSettings(RunSettings defaultSettings, Selector[] selectors) {
+    List<TestSelector> testSelectors =
+        Arrays.stream(selectors)
+            .flatMap(
+                selector -> {
+                  if (selector instanceof TestSelector) {
+                    return Stream.of((TestSelector) selector);
+                  } else {
+                    return Stream.empty();
+                  }
+                })
+            .toList();
```

Comment:
@olafurpg this is the best what can I do in Java 😅 At least now, casting is done just after `instanceof` check

Reply:
LGTM 👍

Reply:
I think you can replace the if/else with a ternary operator to drop the block expression with return statements